### ISSUE TITLE
Standard bookmarks (documents etc) not renamed after user language change

### ISSUE THREAD

**Issue by mcclurgm** [elementary/files#1818]
## What Happened

I recently changed my user language (not system) from English to German. When I opened Files, it kindly offered to rename my Documents, Downloads, and such folders from English to German, which I did. This worked perfectly, but the sidebar bookmarks are still in English. They correctly correspond to the German-named folders, but they're just displayed with English names. I'm not sure if this is a missing translation issue or if it's something else. Here's a screenshot of my Music folder (Musik in German, which you can see in the path bar):
![Bildschirmfoto von 2021-08-28 16 33 25](https://user-images.githubusercontent.com/29022469/131232584-b74dc357-367c-40cb-ace4-42488b239b91.png)

This may be related to #1810, but the behavior I'm seeing here is almost the opposite of what that issue describes.

## Expected Behavior

Bookmark names should be displayed in the correct language

## Steps to Reproduce

1. Change user language from English (to German, to replicate my steps exactly)
2. When Files prompts to rename directories, accept
3. Open Files and see English names in the sidebar

## Platform Information

```
(io.elementary.files:2380): GLib-GIO-DEBUG: 16:36:44.090: _g_io_module_get_default: Found default implementation dconf (DConfSettingsBackend) for ‘gsettings-backend’
(io.elementary.files:2380): dconf-DEBUG: 16:36:44.090: watch_fast: "/io/elementary/files/preferences/" (establishing: 0, active: 0)
(io.elementary.files:2380): dconf-DEBUG: 16:36:44.090: watch_fast: "/io/elementary/files/icon-view/" (establishing: 0, active: 0)
(io.elementary.files:2380): dconf-DEBUG: 16:36:44.090: watch_fast: "/io/elementary/files/list-view/" (establishing: 0, active: 0)
(io.elementary.files:2380): dconf-DEBUG: 16:36:44.090: watch_fast: "/io/elementary/files/column-view/" (establishing: 0, active: 0)
(io.elementary.files:2380): GLib-DEBUG: 16:36:44.090: unsetenv() is not thread-safe and should not be used after threads are created
(io.elementary.files:2380): dconf-DEBUG: 16:36:44.090: watch_fast: "/org/gnome/desktop/interface/" (establishing: 0, active: 0)
(io.elementary.files:2380): dconf-DEBUG: 16:36:44.090: watch_fast: "/org/gnome/desktop/privacy/" (establishing: 0, active: 0)
(io.elementary.files:2380): dconf-DEBUG: 16:36:44.090: watch_fast: "/org/gtk/settings/file-chooser/" (establishing: 0, active: 0)
(io.elementary.files:2380): dconf-DEBUG: 16:36:44.092: watch_established: "/io/elementary/files/preferences/" (establishing: 1)
(io.elementary.files:2380): dconf-DEBUG: 16:36:44.092: watch_established: "/io/elementary/files/icon-view/" (establishing: 1)
(io.elementary.files:2380): dconf-DEBUG: 16:36:44.092: watch_established: "/io/elementary/files/list-view/" (establishing: 1)
(io.elementary.files:2380): dconf-DEBUG: 16:36:44.092: watch_established: "/io/elementary/files/column-view/" (establishing: 1)
(io.elementary.files:2380): dconf-DEBUG: 16:36:44.092: watch_established: "/org/gnome/desktop/interface/" (establishing: 1)
(io.elementary.files:2380): dconf-DEBUG: 16:36:44.092: watch_established: "/org/gnome/desktop/privacy/" (establishing: 1)
(io.elementary.files:2380): dconf-DEBUG: 16:36:44.092: watch_established: "/org/gtk/settings/file-chooser/" (establishing: 1)
(io.elementary.files:2380): GLib-GIO-DEBUG: 16:36:44.100: _g_io_module_get_default: Found default implementation gvfs (GDaemonVfs) for ‘gio-vfs’
(io.elementary.files:2380): GLib-DEBUG: 16:36:44.127: unsetenv() is not thread-safe and should not be used after threads are created
(io.elementary.files:2380): Gtk-DEBUG: 16:36:44.127: Connecting to session manager
** (io.elementary.files:2380): DEBUG: 16:36:44.129: PluginManager.vala:130: Loading plugin for /usr/lib/x86_64-linux-gnu/io.elementary.files/plugins/core/libpantheon-files-cloudproviders.so
** (io.elementary.files:2380): DEBUG: 16:36:44.130: PluginManager.vala:157: Loaded module source: '/usr/lib/x86_64-linux-gnu/io.elementary.files/plugins/core/libpantheon-files-cloudproviders.so'
** (io.elementary.files:2380): DEBUG: 16:36:44.130: PluginManager.vala:130: Loading plugin for /usr/lib/x86_64-linux-gnu/io.elementary.files/plugins/core/libpantheon-files-contractor.so
** (io.elementary.files:2380): DEBUG: 16:36:44.130: PluginManager.vala:157: Loaded module source: '/usr/lib/x86_64-linux-gnu/io.elementary.files/plugins/core/libpantheon-files-contractor.so'
** (io.elementary.files:2380): DEBUG: 16:36:44.130: PluginManager.vala:130: Loading plugin for /usr/lib/x86_64-linux-gnu/io.elementary.files/plugins/core/libpantheon-files-ctags.so
** (io.elementary.files:2380): DEBUG: 16:36:44.132: PluginManager.vala:157: Loaded module source: '/usr/lib/x86_64-linux-gnu/io.elementary.files/plugins/core/libpantheon-files-ctags.so'
** (io.elementary.files:2380): DEBUG: 16:36:44.132: PluginManager.vala:130: Loading plugin for /usr/lib/x86_64-linux-gnu/io.elementary.files/plugins/core/libpantheon-files-git.so
** (io.elementary.files:2380): DEBUG: 16:36:44.137: PluginManager.vala:157: Loaded module source: '/usr/lib/x86_64-linux-gnu/io.elementary.files/plugins/core/libpantheon-files-git.so'
** (io.elementary.files:2380): DEBUG: 16:36:44.137: PluginManager.vala:130: Loading plugin for /usr/lib/x86_64-linux-gnu/io.elementary.files/plugins/core/libpantheon-files-trash.so
** (io.elementary.files:2380): DEBUG: 16:36:44.141: PluginManager.vala:157: Loaded module source: '/usr/lib/x86_64-linux-gnu/io.elementary.files/plugins/core/libpantheon-files-trash.so'
** Message: 16:36:44.172: Application.vala:133: Report any issues/bugs you might find to https://github.com/elementary/files/issues
** (io.elementary.files:2380): DEBUG: 16:36:44.312: Directory.vala:1143: Dir file:///home/michael/Downloads not in cache
** (io.elementary.files:2380): DEBUG: 16:36:44.319: Directory.vala:196: Preparing directory for loading
** (io.elementary.files:2380): DEBUG: 16:36:44.319: Directory.vala:231: get_file_info
** (io.elementary.files:2380): DEBUG: 16:36:44.319: Directory.vala:262: try_query_info
** (io.elementary.files:2380): DEBUG: 16:36:44.320: Directory.vala:287: got file info - updating
** (io.elementary.files:2380): DEBUG: 16:36:44.320: File.vala:491: Enthaltender Einhängepunkt für Datei /home/michael/Downloads/Temporary wurde nicht gefunden
** (io.elementary.files:2380): DEBUG: 16:36:44.320: Directory.vala:289: success true; enclosing mount null
** (io.elementary.files:2380): DEBUG: 16:36:44.320: File.vala:491: Enthaltender Einhängepunkt für Datei /home/michael/Downloads/Temporary wurde nicht gefunden
** (io.elementary.files:2380): DEBUG: 16:36:44.320: Directory.vala:223: success true; enclosing mount null
** (io.elementary.files:2380): DEBUG: 16:36:44.320: Directory.vala:422: make ready
** (io.elementary.files:2380): DEBUG: 16:36:44.320: Directory.vala:611: list directory async
** (io.elementary.files:2380): DEBUG: 16:36:44.321: Directory.vala:641: Obtained file enumerator for location file:///home/michael/Downloads/Temporary
** (io.elementary.files:2380): DEBUG: 16:36:44.323: Directory.vala:1143: Dir file:///home/michael/Downloads not in cache
** (io.elementary.files:2380): DEBUG: 16:36:44.323: Directory.vala:1143: Dir file:///home/michael not in cache
** (io.elementary.files:2380): DEBUG: 16:36:44.323: Directory.vala:196: Preparing directory for loading
** (io.elementary.files:2380): DEBUG: 16:36:44.323: Directory.vala:231: get_file_info
** (io.elementary.files:2380): DEBUG: 16:36:44.323: Directory.vala:262: try_query_info
** (io.elementary.files:2380): DEBUG: 16:36:44.325: Directory.vala:1143: Dir file:///home/michael not in cache
** (io.elementary.files:2380): DEBUG: 16:36:44.325: Directory.vala:1143: Dir file:///home/michael not in cache
** (io.elementary.files:2380): DEBUG: 16:36:44.325: Directory.vala:1143: Dir file:///home/michael not in cache
** (io.elementary.files:2380): DEBUG: 16:36:44.325: Directory.vala:1143: Dir file:///home/michael not in cache
** (io.elementary.files:2380): DEBUG: 16:36:44.325: Directory.vala:1143: Dir file:///home/michael not in cache
** (io.elementary.files:2380): DEBUG: 16:36:44.326: Directory.vala:1143: Dir file:///home not in cache
** (io.elementary.files:2380): DEBUG: 16:36:44.326: File.vala:491: Enthaltender Einhängepunkt für Datei /home/michael wurde nicht gefunden
** (io.elementary.files:2380): DEBUG: 16:36:44.328: File.vala:491: Es wurde kein gültiger Speicherort gefunden
** (io.elementary.files:2380): DEBUG: 16:36:44.329: Directory.vala:1143: Dir file:///home/michael not in cache
** (io.elementary.files:2380): DEBUG: 16:36:44.329: File.vala:491: Enthaltender Einhängepunkt für Datei /home/michael/Projekte wurde nicht gefunden
** (io.elementary.files:2380): DEBUG: 16:36:44.330: Directory.vala:1143: Dir file:///home/michael not in cache
** (io.elementary.files:2380): DEBUG: 16:36:44.331: File.vala:491: Enthaltender Einhängepunkt für Datei /home/michael/Musik wurde nicht gefunden
** (io.elementary.files:2380): DEBUG: 16:36:44.331: Directory.vala:1143: Dir file:///home/michael not in cache
** (io.elementary.files:2380): DEBUG: 16:36:44.332: File.vala:491: Enthaltender Einhängepunkt für Datei /home/michael/Bilder wurde nicht gefunden
** (io.elementary.files:2380): DEBUG: 16:36:44.332: Directory.vala:1143: Dir file:///home/michael not in cache
** (io.elementary.files:2380): DEBUG: 16:36:44.332: File.vala:491: Enthaltender Einhängepunkt für Datei /home/michael/Videos wurde nicht gefunden
** (io.elementary.files:2380): DEBUG: 16:36:44.333: Directory.vala:1143: Dir file:///home/michael not in cache
** (io.elementary.files:2380): DEBUG: 16:36:44.333: File.vala:491: Enthaltender Einhängepunkt für Datei /home/michael/Downloads wurde nicht gefunden
** (io.elementary.files:2380): DEBUG: 16:36:44.334: File.vala:491: Es wurde kein gültiger Speicherort gefunden
** (io.elementary.files:2380): DEBUG: 16:36:44.336: Directory.vala:287: got file info - updating
** (io.elementary.files:2380): DEBUG: 16:36:44.336: File.vala:491: Enthaltender Einhängepunkt für Datei /home/michael/Downloads wurde nicht gefunden
** (io.elementary.files:2380): DEBUG: 16:36:44.336: Directory.vala:289: success true; enclosing mount null
** (io.elementary.files:2380): DEBUG: 16:36:44.336: File.vala:491: Enthaltender Einhängepunkt für Datei /home/michael/Downloads wurde nicht gefunden
** (io.elementary.files:2380): DEBUG: 16:36:44.336: Directory.vala:223: success true; enclosing mount null
** (io.elementary.files:2380): DEBUG: 16:36:44.336: Directory.vala:422: make ready
** (io.elementary.files:2380): DEBUG: 16:36:44.336: Directory.vala:611: list directory async
** (io.elementary.files:2380): DEBUG: 16:36:44.336: File.vala:491: Enthaltender Einhängepunkt für Datei /home/michael/Videos wurde nicht gefunden
** (io.elementary.files:2380): DEBUG: 16:36:44.337: File.vala:491: Enthaltender Einhängepunkt für Datei /home/michael/Downloads wurde nicht gefunden
** (io.elementary.files:2380): DEBUG: 16:36:44.337: File.vala:491: Enthaltender Einhängepunkt für Datei /home/michael/Bilder wurde nicht gefunden
** (io.elementary.files:2380): DEBUG: 16:36:44.337: File.vala:491: Enthaltender Einhängepunkt für Datei /home/michael/Musik wurde nicht gefunden
** (io.elementary.files:2380): DEBUG: 16:36:44.337: File.vala:491: Enthaltender Einhängepunkt für Datei /home/michael/Projekte wurde nicht gefunden
** (io.elementary.files:2380): DEBUG: 16:36:44.337: CallWhenReady.vala:87: call when ready OK - empty list
** (io.elementary.files:2380): DEBUG: 16:36:44.337: Directory.vala:641: Obtained file enumerator for location file:///home/michael/Downloads
** (io.elementary.files:2380): DEBUG: 16:36:44.337: File.vala:491: Enthaltender Einhängepunkt für Datei /home/michael/Downloads/Temporary/Bildschirmfoto von 2021-08-28 16.33.25.png wurde nicht gefunden
** (io.elementary.files:2380): DEBUG: 16:36:44.337: File.vala:491: Enthaltender Einhängepunkt für Datei /home/michael/Downloads/Mathematica_12.3.1_LINUX.sh.part wurde nicht gefunden
** (io.elementary.files:2380): DEBUG: 16:36:44.338: File.vala:491: Enthaltender Einhängepunkt für Datei /home/michael/Downloads/Mathematica_12.3.1_LINUX.sh wurde nicht gefunden
** (io.elementary.files:2380): DEBUG: 16:36:44.338: File.vala:491: Enthaltender Einhängepunkt für Datei /home/michael/Downloads/Temporary wurde nicht gefunden
** (io.elementary.files:2380): DEBUG: 16:36:44.338: Directory.vala:725: after loading state is FILES_DIRECTORY_STATE_LOADED
** (io.elementary.files:2380): DEBUG: 16:36:44.339: plugin.vala:159: Error opening git repository at file:///home/michael/Downloads/Temporary: could not find repository from '/home/michael/Downloads/Temporary'
(io.elementary.files:2380): dconf-DEBUG: 16:36:44.339: change_fast
(io.elementary.files:2380): dconf-DEBUG: 16:36:44.339: change_notify: /io/elementary/files/preferences/default-viewmode
** (io.elementary.files:2380): DEBUG: 16:36:44.341: Directory.vala:725: after loading state is FILES_DIRECTORY_STATE_LOADED
** (io.elementary.files:2380): DEBUG: 16:36:44.371: Directory.vala:529: Directory is last toggle_ref_notify file:///home/michael/Downloads
** (io.elementary.files:2380): DEBUG: 16:36:44.371: Directory.vala:153: Directory destruct file:///home/michael/Downloads
** (io.elementary.files:2380): DEBUG: 16:36:44.407: Thumbnailer.vala:290: started 6
** (io.elementary.files:2380): DEBUG: 16:36:44.407: Directory.vala:1130: found cached dir file:///home/michael/Downloads/Temporary
** (io.elementary.files:2380): DEBUG: 16:36:44.546: Directory.vala:1143: Dir file:///home not in cache
** (io.elementary.files:2380): DEBUG: 16:36:44.550: Directory.vala:1143: Dir file:///home/michael not in cache
** (io.elementary.files:2380): DEBUG: 16:36:44.551: Directory.vala:1143: Dir file:///home/michael not in cache
** (io.elementary.files:2380): DEBUG: 16:36:44.553: Directory.vala:1143: Dir file:///home/michael not in cache
** (io.elementary.files:2380): DEBUG: 16:36:44.554: Directory.vala:1143: Dir file:///home/michael not in cache
** (io.elementary.files:2380): DEBUG: 16:36:44.556: Directory.vala:1143: Dir file:///home/michael not in cache
** (io.elementary.files:2380): DEBUG: 16:36:44.558: File.vala:1065: Vorgang wird nicht unterstützt
** (io.elementary.files:2380): DEBUG: 16:36:44.561: Directory.vala:1143: Dir file:///home not in cache
** (io.elementary.files:2380): DEBUG: 16:36:44.563: Directory.vala:1143: Dir file:///home not in cache
```

**Comment by jeremypw** [comment]
It is the system, not Files, that offers to rename the home folders and does it, including updating the uris in the bookmark file. However, the "custom names" are not changed. By default, the custom name is the same as the basename.  The custom name field should be blank in this case but it looks like since the Sidebar rewrite, the bookmarks are being created with the custom name filled in with the basename, so I think this is a regression.  Thanks for spotting this.

**Comment by mcclurgm** [comment]
I see you have an open PR for this, but a bit more info I found. The `~/.config/gtk+-3.0/bookmarks` directory keeps the English names, and I imagine that's where Files got the name from. When I try to change the name to German and open Files, each entry I changed gets a duplicate item where one is English and one is German, so it seems like Files is trying to modify this with its idea of the names.

When I delete and re-add the bookmarks, they use the proper German names now. So it seems like the issue is cached data of some sort, and re-adding them clears that data to make room for the updated names.

**Comment by jeremypw** [comment]
There is no cached bookmark data that is maintained by Files separately from the `bookmarks` file once Files closes. So I am not sure where the two items with different custom names came from :shrug:. I think in master, Files regards two bookmarks with the same uri but different custom name as different but the pending PRs will only allow one bookmark per uri. You are right that if the `bookmarks` file is deleted it gets rebuilt by Files from the `GLib.UserDirectory` enum and `GLib.Environment.get_user_special_dir ()`.

**Comment by mcclurgm** [comment]
Interesting. In that case I'm not sure how I wound up with duplicate entries. Maybe some other service tried to mess with them. 🤷🏻 

Some things to clear up. I'm not sure if we're on the same page or if I wasn't clear, so I'll write them down here:

The only time that I saw two different entries in the bookmarks file was after I manually edited the file, not when I was editing them through the Files GUI. I only saw one bookmark entry each in Files even with the duplicate entries in the bookmarks file.

When I said delete and re-add bookmarks, I meant individually from the Files interface, not by deleting the bookmarks file. I removed the bookmark from the sidebar, then dragged it back out, and it used the new name. I imagine that's the same result that you're describing (deleted and rebuilt), but in case that makes a difference.
